src/GTP.cpp

```diff
@@ -353,7 +360,7 @@ bool GTP::execute(GameState & game, std::string xinput) {
             if (cfg_allow_pondering) {
                 // now start pondering
                 if (!game.has_resigned()) {
-                    auto search = std::make_unique<UCTSearch>(game);
+                    search->set_gamestate(game);
```

Comment:
Does it not make more sense to have think() and ponder() take the gamestate?

Reply:
fixed.